### ISSUE TITLE
[Accessibility]: Commits page - Copy SHA button does not inform screen readers of status

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#34973]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34423

### Problem description

Button state 1:
<img width="195" alt="image" src="https://user-images.githubusercontent.com/9516420/166908968-064261dc-84bf-4a62-b63d-4e99f484cab0.png">

Button state 2:
<img width="182" alt="image" src="https://user-images.githubusercontent.com/9516420/166909044-af161330-1ffa-4861-b848-fbaf01fcdcd1.png">

This information is useful for visual users but not supported for screen readers.

### Expected behavior

We should use `screenReaderAnnounce` to quickly inform users when this state changes

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
